store/store.go

```diff
@@ -5,64 +5,40 @@ import (
 	"time"
 )
 
-type ChangeEventType int
-
-const (
-	Created ChangeEventType = iota
-	Updated
-)
-
-func (c ChangeEventType) String() string {
-	switch c {
-	case Created:
-		return "Created"
-	case Updated:
-		return "Updated"
-	}
-	return "unknown"
-}
-
 var (
-	// ErrSecretNotFound is returned if the specified secret is not found in the
-	// parameter store
-	ErrSecretNotFound = errors.New("secret not found")
+	// ErrSecretsNotFound is returned if the specified Secrets is not found
+	ErrSecretsNotFound = errors.New("secrets not found")
 )
 
-type SecretId struct {
-	Service string
-	Key     string
-}
-
-type Secret struct {
-	Value *string
-	Meta  SecretMetadata
+type Secrets struct {
+	Secrets RawSecrets
+	Meta    *SecretsMetadata
 }
 
 // A secret without any metadata
-type RawSecret struct {
-	Value string
-	Key   string
-}
+type RawSecrets map[string]string
 
-type SecretMetadata struct {
-	Created   time.Time
-	CreatedBy string
-	Version   int
-	Key       string
+func (r *RawSecrets) Validate() error {
+	// TODO; need to move validateKey here
+	panic("not implemented")
```

Comment:
just making sure you know this still exists :)

Reply:
Fixed